kubetest2-gke/deployer/build.go

```diff
@@ -47,3 +52,31 @@ func (d *deployer) verifyBuildFlags() error {
 	d.BuildOptions.RepoRoot = d.RepoRoot
 	return d.BuildOptions.Validate()
 }
+
+// ensure that the version is a valid gke version
+func normalizeVersion(version string) (string, error) {
+	// (1.18.10)(-gke)(.601.123)(+existingSuffix)
+	// capture repeated occurrences of group 3 and don't capture just last
+	re, err := regexp.Compile(`(\d\.\d+\.\d*)(-\w+)?((?:\.\d+)*)(\+.*)*`)
```

Comment:
Worth forcing as a constant with [mustCompile](https://golang.org/pkg/regexp/#MustCompile)? I forget where but someone had me use that on one of my PRs and will fail fast if a change breaks the regex.

Reply:
done